section3/reflection.md

```diff
@@ -2,16 +2,31 @@
+
 3. What is a Hash, and how is it different from an Array?
 
+  * A Hash is different from arrays as it allows you to set keys to the values that are within the Hash. Where as in an array you can only have multiple or no values, a Hash will require a key to also use for the values that are within it. This allows the user to pull either the keys or values when looking for the information within the Hash's where as you only had one layer of information within an array. This allows for users to find different methods of pulling multiple data points.
+
 4. In the space below, create a Hash stored to a variable named `pet_store`.  This hash should hold an inventory of items and the number of that item that you might find at a pet store.
+  * `pet store = {dog_food: 5, cat_food: 8, fish_food: 15, dog_toys 8, cat_toys: 10}`
 
 5. Given the following `states = {"CO" => "Colorado", "IA" => "Iowa", "OK" => "Oklahoma"}`, how would you access the value `"Iowa"`?
+  * `p states.fetch("IA")`
 
 6. With the same hash above, how would we get all the keys?  How about all the values?
+  * `p states.keys` for finding keys of `states`
+  * `p states.values` for finding values of `states`
 
 7. What is another example of when we might use a hash?  In your example, why is a hash better than an array?
 
-8. What questions do you still have about hashes?
\ No newline at end of file
+  * A hash is better than an array and is used in order to have values that are tied to different values, and can be pulled either way through the key or value. This is much more efficient when creating data lists with more than one variable. This means you can now create data hashes including multiple peieces of information for on variable key. This allows users to pull very specific information for different keys.
+
+8. What questions do you still have about hashes?
+  * When using an hash with possibly hundreds of of different variables and keys, can one list these in a more viewable way, or would one create more hashes to include the excess data?
+  * Can hashes include endless key variable paired relationship's as in they can have multiple variables tied to a key, which show different data about each of the keys?
```

Comment:
Hmmm I'm not entirely sure I follow...

You could have a key that has a value of an array which holds multiple variables.